BLSQ/iaso
# POLIO-1275 [Line-list] Polio notifications import backend

Backend models for Polio line-list: allow to import (sync or async) and store polio notifications.

Related JIRA tickets : [POLIO-1275](https://bluesquare.atlassian.net/browse/POLIO-1275)

## Changes

- New models in `polio`
    - `Notification`
    - `NotificationImport`
- New `NotificationXlsxImporter` class
- New async task to import notifications
- Moved `parents_q` (since it's used multiple times):
    - from `plugins/polio/api/dashboards/forma.py`
    - to `iaso/models/org_unit.py` (in a custom `OrgUnitManager`)

## Prerequisites

You **must** add the following `OrgUnit` aliases for the import to work:

- `DEMOCRATIC REPUBLIC OF THE CONGO` -> `RDC`
- `CÔTE D’IVOIRE` -> `CIV`
- `CENTRAL AFRICAN REPUBLIC` -> `CAR`
- `GUINEA-BISSAU` -> `GUINEA BISSAU`
- `UNITED REPUBLIC OF TANZANIA` -> `TANZANIA`

## How to test / Video

- check the prerequisites above
- ensure that a background worker is running
    - `docker-compose run iaso manage tasks_worker`
- grab the `.xlsx` file [from the Jira ticket's comments](https://bluesquare.atlassian.net/browse/POLIO-1275?focusedCommentId=59457)
- go to `http://localhost:8081/admin/polio/notificationimport/`
- create a new `NotificationImport` for the `.xlsx` file
- select `Create notifications` from the `Action` menu
    - an async task will be created
- go to `http://localhost:8081/admin/polio/notification/` to see the result

See the video:

https://github.com/BLSQ/iaso/assets/281139/7dad5eb8-2762-4c69-8a47-df9f51ca774f

[POLIO-1275]: https://bluesquare.atlassian.net/browse/POLIO-1275?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
